### PR TITLE
adding identityserver4 ref

### LINKED ISSUE #1849: Add links to IdentityServer4 quick starts
Under security -> authentication add:

- Securing Web APIs with IdentityServer4 (identityserver/index.rst)
  - [Overview](https://identityserver4.readthedocs.io/en/dev/quickstarts/0_overview.html)
  - [Protecting an API using client credentials](https://identityserver4.readthedocs.io/en/dev/quickstarts/1_client_credentials.html)
  - [Protecting an API using Passwords](https://identityserver4.readthedocs.io/en/dev/quickstarts/2_resource_owner_passwords.html)
  - [Adding user authentication with OpenID Connect](https://identityserver4.readthedocs.io/en/dev/quickstarts/3_interactive_login.html)
  - [Adding support for external authentication](https://identityserver4.readthedocs.io/en/dev/quickstarts/4_external_authentication.html)
  - [Hybrid flow and API access](https://identityserver4.readthedocs.io/en/dev/quickstarts/5_hybrid_and_api_access.html)
  - [Using with ASP.NET Core Identity](https://identityserver4.readthedocs.io/en/dev/quickstarts/6_aspnet_identity.html)
  - [Adding a JavaScript client](https://identityserver4.readthedocs.io/en/dev/quickstarts/7_javascript_client.html)

### PR DESCRIPTION
**Pull request by spboyer**
fixes #1849

### CODE REVIEW

**Comment by spboyer** [comment]
@danroth27

File: aspnet/security/authentication/index.rst [hunk 1]
```diff
@@ -11,3 +11,4 @@ Authentication
   oauth2
   cookie
   azure-active-directory/index
+  indentityserver/index
```
Comment (danroth27) [inline aspnet/security/authentication/index.rst]: indentityserver -> identityserver :smile:

File: aspnet/security/authentication/indentityserver/index.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Securing Web APIs with IdentityServer4
+======================================
+
+.. toctree::
+  :titlesonly:
+
+  Overview<https://identityserver4.readthedocs.io/en/dev/quickstarts/0_overview.html>
```
Comment (danroth27) [inline aspnet/security/authentication/indentityserver/index.rst]: I think typically we put a space between the friendly text and the link in the angle brackets.

**Comment by danroth27** [comment]
@brockallen @leastprivilege

**Comment by danroth27** [comment]
:watch:

**Comment by brockallen** [comment]
We'll have one more link by tomorrow for EF, but I can send a PR once it's done.

**Comment by danroth27** [comment]
Should we consider just linking off directly to https://identityserver4.readthedocs.io/ instead of trying to list all of the individual quickstarts? We could make the link under security -> authentication a bit more generic: [Security ASP.NET Core apps with identityServer4](https://identityserver4.readthedocs.io/) or something like that.

**Comment by brockallen** [comment]
I'll defer to @leastprivilege on that -- he'll have a stronger opinion than me.

**Comment by leastprivilege** [comment]
I definitely like the broader title using "ASP.NET Core Applications" - maybe list the fundamental quickstarts (so people know what to expect) and also do a link to the our docs main page.

**Comment by danroth27** [comment]
Which quickstarts would you consider fundamental? 

I still lean towards just linking directly from security -> authentication to the IdentityServer4 docs. That way we don't have to keep our links in sync with yours and you have full control over how folks get introduced to your framework.

**Comment by leastprivilege** [comment]
OK - fine with me.

**Comment by spboyer** [comment]
So I think I am reading that the decision is to link the TOC item directly to  https://identityserver4.readthedocs.io/ instead of having a identityserver/index correct?

@danroth27

**Comment by danroth27** [comment]
Yeah, we can just add a TOC entry to security/authentication/index.rst:

`Securing ASP.NET Core apps with identityServer4 <https://identityserver4.readthedocs.io>`

**Comment by brockallen** [comment]
`IdentityServer4`

**Comment by spboyer** [comment]
ok, check it.

**Comment by danroth27** [comment]
:shipit:

**Pull request merged**